test/rest.browser.test.js

```diff
@@ -300,6 +300,185 @@ describe('strong-remoting-rest', function(){
+                returns: [
+                  { arg: 'enumParam', type: 'string'}
+                ],
+                http: { path: '/' }
+              }
+          );
+          objects.invoke(method.name, ['d'], function(err, a) {
+            expect(err).to.be.an.instanceof(Error);
+            expect(err.message).to.equal("'d' is not one of: a, b, c");
+            done();
+          });
+        });
+        it(', should allow missing not required enum values', function(done) {
+          var method = givenSharedStaticMethod(
+              function bar(enumParam, cb) {
+                cb(null, enumParam);
+              },
+              {
+                accepts: [
+                  { arg: 'enumParam', type: 'string', required: false, enum: ['a', 'b', 'c']}
+                ],
+                returns: [
+                  { arg: 'enumParam', type: 'string'}
+                ],
+                http: { path: '/' }
+              }
+          );
+          objects.invoke(method.name, [null], function(err, a) {
+            expect(err).to.not.exist;
+            expect(a).to.not.exist;
+            done();
+          });
+        });
+        it(', should match all parameters to a value in enum', function(done) {
+          var method = givenSharedStaticMethod(
+              function bar(enumParam, cb) {
+                cb(null, enumParam);
+              },
+              {
+                accepts: [
+                  { arg: 'enumParam', type: 'string', required: false, enum: ['a', 'b', 'c']}
+                ],
+                returns: [
+                  { arg: 'enumParam', type: 'string'}
+                ],
+                http: { path: '/' }
+              }
+          );
+          objects.invoke(method.name, ['a'], function(err, a) {
+            expect(err).to.not.exist;
+            expect(a).to.exist;
+            assert.equal(['a', 'b', 'c'].indexOf(a) > -1, true);
+            done();
+          });
+        });
+        it(', should match all parameters to all values in enum', function(done) {
+          var method = givenSharedStaticMethod(
+              function bar(enumParam, cb) {
+                cb(null, enumParam);
+              },
+              {
+                accepts: [
+                  { arg: 'enumParam', type: ['string'], required: false, enum: ['a', 'b', 'c']}
+                ],
+                returns: [
+                  { arg: 'enumParam', type: ['string']}
+                ],
+                http: { path: '/' }
+              }
+          );
+          objects.invoke(method.name, [['a', 'b', 'c']], function(err, a) {
+            expect(err).to.not.exist;
+            expect(a).to.exist;
+            expect(a).to.eql(['a', 'b', 'c']);
+            done();
+          });
+        });
+      });
+     describe('under items', function () {
+       it(', should not allow parameters that are required and not found in enum', function(done) {
+         var method = givenSharedStaticMethod(
```

Comment:
Can you move this method definition out of `it`, and reuse it for all of the `it`s within this `describe`?